mail_template_conditional_attachment/models/mail_template.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Druidoo - Iván Todorovich <ivan.todorovich@druidoo.io>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields
+from odoo.tools import pycompat
+
+
+class MailTemplate(models.Model):
+    _inherit = 'mail.template'
+
+    conditional_attachment_ids = fields.One2many(
+        'mail.template.conditional.attachment',
+        'mail_template_id',
+        string='Conditional Attachments',
+    )
+
+    @api.multi
+    def generate_email(self, res_ids, **kwargs):
+        """ Overload this method to attach conditional attachments """
+        self.ensure_one()
+
+        # Odoo way of handling different result for multi/one res_ids
+        # Placed before super(), so we force multi_mode=True there
+        multi_mode = True
+        if isinstance(res_ids, pycompat.integer_types):
+            res_ids = [res_ids]
+            multi_mode = False
+
+        results = super().generate_email(res_ids, **kwargs)
+
+        # Add conditional attachments
+        if self.conditional_attachment_ids:
```

Comment:
this method is multi: here it would be broken if `len(self) > 1`.
I'd do:
```py
cond_attachments = {
    x.id: conditional_attachment_ids.get_attachment_ids(x.id)
    for x in self if x.conditional_attachment_ids
}
```
then use this mapping in the loop (if valued at all).

Reply:
Actually this method is meant to be called only on one template. That's why there's `self.ensure_one()` at the beggining.

https://github.com/odoo/odoo/blob/12.0/addons/mail/models/mail_template.py#L422

The `multi_mode` mentioned before is because the template may generate more than 1 email. But the template will always be 1. It shouldn't be confused with `api.multi`

Reply:
haha my bad, I totally missed the ensure_one call :sweat_smile: